### PR TITLE
[ENH] Adding `VAR._predict_intervals`

### PR DESCRIPTION
**Pull request by lbventura**
#### Reference Issues/PRs

Adds `predict_intervals` to the VAR module.

#### What does this implement/fix? Explain your changes.

Adds `predict_intervals` to the VAR module.

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/alan-turing-institute/sktime/blob/main/.all-contributorsrc).
- [ ] Optionally, I've updated sktime's [CODEOWNERS](https://github.com/alan-turing-institute/sktime/blob/main/CODEOWNERS) to receive notifications about future changes to these files.
- [ ] I've added unit tests and made sure they pass locally.
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG] indicating whether the PR topic is related to enhancement, maintenance, documentation, or bug.

##### For new estimators
- [ ] I've added the estimator to the online documentation.
- [ ] I've updated the existing example notebooks or provided a new one to showcase how my estimator works.

### CODE REVIEW

File: sktime/forecasting/var.py [hunk 1]
```diff
@@ -184,6 +185,112 @@ def _predict(self, fh, X=None):
...
+        pre_output_df = concat_df[list(concat_df_columns)]
+
+        pre_output_df_2 = pd.DataFrame(
+            pre_output_df.values,
+            columns=pd.MultiIndex.from_tuples(
+                [col.split(" ") for col in pre_output_df]
+            ),
+        )
+
+        final_columns = [
```
Comment (fkiraly) [inline sktime/forecasting/var.py:277]: ising `zip` might be more elegant than a loop?

File: sktime/forecasting/var.py [hunk 2]
```diff
@@ -184,6 +185,112 @@ def _predict(self, fh, X=None):
...
+        steps = fh_int[-1]
+        n_lags = model.k_ar
+
+        y_cols_no_space = [str(col).replace(" ", "") for col in self._y.columns]
+
+        df_list = []
+
+        for cov in coverage:
+
+            alpha = -0.5 * cov + 0.5
```
Comment (fkiraly) [inline sktime/forecasting/var.py:237]: the docs say "significance level", so I think we should have `alpha = 1 - cov`, instead

**Review by fkiraly — CHANGES_REQUESTED** [review]
Nice, great work!

Small but blocking detail: I think your formula for `alpha` is wrong.

According to the `statsmodels` docs, `alpha` must be the significance level (often denoted `p`), and not the quantile point of the lower bound, which is the formula you used.

File: sktime/forecasting/var.py (hunk 2 above)
Comment (lbventura) [reply sktime/forecasting/var.py:237]: Corrected in the upcoming PR

File: sktime/forecasting/var.py (hunk 1 above)
Comment (lbventura) [reply sktime/forecasting/var.py:277]: I am not sure how you suggest using `zip`, wouldn't `zip` just compute the internal product of the three lists (?) when we actually want the Cartesian product. I did something neater (but more complicated and needs a package import) in the upcoming PR.

**Comment by lbventura** [comment]
> Nice, great work!
> 
> Small but blocking detail: I think your formula for `alpha` is wrong.
> 
> According to the `statsmodels` docs, `alpha` must be the significance level (often denoted `p`), and not the quantile point of the lower bound, which is the formula you used.

Thanks for highlighting this, I should have checked this the first place. I went back today and confirmed this was in fact the case.

File: sktime/forecasting/var.py [hunk 3]
```diff
@@ -184,6 +188,126 @@ def _predict(self, fh, X=None):
...
+        pre_output_df = concat_df[concat_df_columns]
+
+        pre_output_df_2 = pd.DataFrame(
+            pre_output_df.values,
+            columns=pd.MultiIndex.from_tuples(
+                [col.split(" ") for col in pre_output_df]
+            ),
+        )
+
+        # final_columns = [
```
Comment (fkiraly) [inline sktime/forecasting/var.py:284]: please remove commented out code

File: sktime/forecasting/var.py [hunk 4]
```diff
@@ -75,6 +78,7 @@ class VAR(_StatsModelsAdapter):
         "requires-fh-in-fit": False,
         "univariate-only": False,
         "ignores-exogeneous-X": False,
```
Comment (fkiraly) [inline sktime/forecasting/var.py:80]: saym is this correct? It does not look like `X` is used inside.

I.e., this is VAR, not VAR-X or VARMAX. Right?

Comment (fkiraly) [reply sktime/forecasting/var.py:80]: while we're here, would also be great if the class docstring could say which statsmodels object it interfaces.

**Review by fkiraly — APPROVED** [review]
Looks good now!

Let me know if/when you want it merged, or whether you want to look after the minor details before that.

File: sktime/forecasting/var.py (hunk 1 above)
Comment (fkiraly) [reply sktime/forecasting/var.py]: ah, you are right, sorry. You want `itertools.product`, not `zip`.

Comment (fkiraly) [reply sktime/forecasting/var.py]: ... which you actually used, all fine then!

**Comment by lbventura** [comment]
> Looks good now!
> 
> Let me know if/when you want it merged, or whether you want to look after the minor details before that.

I'll handle these tonight and let you know  👏

**Review by fkiraly — APPROVED** [review]
Great, thanks!

**Pull request merged**